### PR TITLE
Adds finger snapping as an emote

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
*snap

## Why It's Good For The Game
image

## Changelog
:cl:
add: Added finger snapping as an emote.
/:cl:

### CODE REVIEW

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -179,6 +179,15 @@
 					to_chat(user, span_notice("You cannot [key] while stunned."))
 					return FALSE
 
+		if(flags_emote & EMOTE_ARMS_CHECK)
+			if(ishuman(user))
```
Comment (BraveMole) [inline code/datums/emotes.dm:183]: that means a non human can snap non-existing fingers :/

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/datums/emotes.dm (hunk 1 above)
Comment (lbnesquik) [reply code/datums/emotes.dm:183]: Fuck non humans

Comment (lbnesquik) [reply code/datums/emotes.dm:183]: fixed

Comment (lbnesquik) [reply code/datums/emotes.dm:183]: Tested too!

File: code/datums/emotes.dm [hunk 2]
```diff
@@ -179,6 +179,15 @@
...
 					return FALSE
 
+		if(flags_emote & EMOTE_ARMS_CHECK)
+			///okay snapper
+			var/mob/living/carbon/snapper = user
+			var/datum/limb/left_hand = snapper.get_limb("l_hand")
+			var/datum/limb/right_hand = snapper.get_limb("r_hand")
+			if((left_hand.limb_status & LIMB_DESTROYED) && (right_hand.limb_status & LIMB_DESTROYED))
+				to_chat(user, span_notice("You cannot [key] without an arm."))
+				return FALSE
```
Comment (TiviPlus) [inline code/datums/emotes.dm:189]: Couldnt a simplemob run this code in theory
also is it possible for right or left hand to be null

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/datums/emotes.dm (hunk 2 above)
Comment (lbnesquik) [reply code/datums/emotes.dm:189]: I have no clue. I just stole this code and pasted it here, really.

Comment (TiviPlus) [reply code/datums/emotes.dm:189]: If you're porting a pr you need to link it in the pr body

Comment (lbnesquik) [reply code/datums/emotes.dm:189]: The code is from another part of our codebase, i just took the noise from polaris

Comment (lbnesquik) [reply code/datums/emotes.dm:189]: image

I stole it from there, in update_icons

Comment (SplinterGP) [reply code/datums/emotes.dm:189]: limbs cant be null because theyre never removed, theyre permanent datums on mob

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**